catalog/registry/catalog-registry-metacard-filter/src/main/java/ddf/catalog/registry/metacard/filter/RegistryQueryDelegate.java

```diff
@@ -0,0 +1,475 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package ddf.catalog.registry.metacard.filter;
+
+import java.util.Date;
+import java.util.List;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.filter.FilterDelegate;
+
+public class RegistryQueryDelegate extends FilterDelegate<Boolean> {
+
+    /*
+    Returns false if this is a query that should not return registry metacards
+    Return true if this is a query that should return registry metacards
+     */
+
+    @Override
+    public Boolean and(List<Boolean> operands) {
+        return operands.stream()
+                .reduce(false, (a, b) -> a || b);
+    }
+
+    @Override
+    public Boolean or(List<Boolean> operands) {
+        return operands.stream()
+                .reduce(false, (a, b) -> a || b);
+    }
+
+    @Override
+    public Boolean not(Boolean operand) {
+        return operand;
+    }
+
+    @Override
+    public Boolean nearestNeighbor(String propertyName, String wkt) {
+        return false;
+    }
+
+    @Override
+    public Boolean include() {
+        return false;
+    }
+
+    @Override
+    public Boolean exclude() {
+        return false;
+    }
+
+    @Override
+    public Boolean propertyIsEqualTo(String propertyName, String pattern, boolean isCaseSensitive) {
```

Comment:
In practice i don't think this will ever happen since we are the ones creating the registry metacards and setting the content type. If we want to handle the flag i would just add an isCaseSensitive? patter : pattern.toLowerCase() in your startsWith call.